src/Raven.Server/Documents/Handlers/Processors/Queries/AbstractQueriesHandlerProcessorForGet.cs

```diff
@@ -96,43 +96,53 @@ public override async ValueTask ExecuteAsync()
                     {
+                        result?.Dispose();
                         HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
                         return;
                     }
-
-                    if (result.NotModified)
-                    {
-                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
-                        return;
-                    }
-
-                    HttpContext.Response.Headers[Constants.Headers.Etag] = CharExtensions.ToInvariantString(result.ResultEtag);
-
-                    long numberOfResults;
-                    long totalDocumentsSizeInBytes;
-                    await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream(), HttpContext.RequestAborted))
+                    catch (Exception)
                     {
-                        result.Timings = indexQuery.Timings?.ToTimings();
-
-                        (numberOfResults, totalDocumentsSizeInBytes) = await writer.WriteDocumentQueryResultAsync(context, result, metadataOnly, WriteAdditionalData(indexQuery, shouldReturnServerSideQuery), token.Token);
-                        await writer.OuterFlushAsync();
+                        result?.Dispose();
+                        throw;
                     }
 
-
-                    QueryMetadataCache.MaybeAddToCache(indexQuery.Metadata, result.IndexName);
-
-                    if (RequestHandler.ShouldAddPagingPerformanceHint(numberOfResults))
+                    using (result)
                     {
-                        RequestHandler.AddPagingPerformanceHint(PagingOperationType.Queries, $"Query ({result.IndexName})",
-                            $"{indexQuery.Metadata.QueryText}\n{indexQuery.QueryParameters}", numberOfResults, indexQuery.PageSize, result.DurationInMs,
-                            totalDocumentsSizeInBytes);
+                        if (result.NotModified)
+                        {
+                            HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                            return;
+                        }
+
+                        HttpContext.Response.Headers[Constants.Headers.Etag] = CharExtensions.ToInvariantString(result.ResultEtag);
+
+                        long numberOfResults;
+                        long totalDocumentsSizeInBytes;
+                        await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream(), HttpContext.RequestAborted))
+                        {
+                            result.Timings = indexQuery.Timings?.ToTimings();
+
+                            (numberOfResults, totalDocumentsSizeInBytes) = await writer.WriteDocumentQueryResultAsync(context, result, metadataOnly,
+                                WriteAdditionalData(indexQuery, shouldReturnServerSideQuery), token.Token);
+                            await writer.MaybeOuterFlushAsync();
```

Comment:
Note that this line has also been changed. (Hasn't been checked for performance separately)